### ISSUE TITLE
Closing bracket not indented correctly in Rust mode

### ISSUE THREAD

**Issue by RawFisher** [redguardtoo/emacs.d#996]
When I am writing rust code, I found closing bracket not indented correctly.

<img width="229" alt="图片" src="https://user-images.githubusercontent.com/26201819/185743228-502daf2e-5bf2-4ddb-9c58-555d4c661311.png">

My emacs version is emacs28.1
I work on macOS

**Comment by redguardtoo** [comment]
Can you reproduce issue when `emacs -Q`?

**Comment by phye** [comment]
Maybe this is a duplication of https://github.com/redguardtoo/emacs.d/issues/975

Seems no updates from upstream, just wonder if they're using emacs 28 to write any code using c derived mode...

**Comment by RawFisher** [comment]
> What's your major-mode?

rust-mode

**Comment by RawFisher** [comment]
> Maybe this is a duplication of #975
> 
> Seems no updates from upstream, just wonder if they're using emacs 28 to write any code using c derived mode...

You are right. Now I found the same problem appear in c mode...Maybe I should use older version of emacs?

**Comment by RawFisher** [comment]
When I delete all config files, and then
1. M-x package-refresh-contents
2. M-x package-install rust-mode
3. create a .rs file
4. enable electric-pair-mode
Closing bracket can indent correctly
![image](https://user-images.githubusercontent.com/26201819/186452943-54db8369-6a36-4644-8c91-408d9700ebb9.png)

**Comment by redguardtoo** [comment]
I can reproduce the issue after removing my emacs.d and follow your steps,
<img width="900" alt="image" src="https://user-images.githubusercontent.com/184553/186594027-fba579a8-c328-4ef1-a388-6f06854e6dcf.png">

**Comment by redguardtoo** [comment]
You could turn off `electric-pair-mode` or run some indent command manually or automatically in `after-save-hook`. 

I run `evil-indent` manually.
